pallets/capacity/src/lib.rs

```diff
@@ -738,3 +746,51 @@ impl<T: Config> Replenishable for Pallet<T> {
 		false
 	}
 }
+
+impl<T: Config> StakingRewardsProvider<T> for Pallet<T> {
+	type AccountId = T::AccountId;
+	type RewardEra = T::RewardEra;
+	type Hash = T::Hash;
+
+	// Calculate the size of the reward pool for the current era, based on current staked token
+	// and the other determined factors of the current economic model
+	// Currently set to 10% of total staked token.
+	fn reward_pool_size() -> Result<BalanceOf<T>, DispatchError> {
+		let current_era_info = CurrentEraInfo::<T>::get();
+		let current_staked =
+			StakingRewardPool::<T>::get(current_era_info.era_index).unwrap_or_default();
+		if current_staked.total_staked_token.is_zero() {
+			return Ok(BalanceOf::<T>::zero())
+		}
+		Ok(current_staked
+			.total_staked_token
+			.checked_div(&BalanceOf::<T>::from(10u8))
+			.unwrap_or_default())
+	}
+
+	// Performs range checks plus a reward calculation based on economic model for the era range
+	// Currently just rewards 1 unit per era for a valid range since there is no history storage
+	fn staking_reward_total(
+		_account_id: T::AccountId,
+		from_era: T::RewardEra,
+		to_era: T::RewardEra,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		let max_eras = T::RewardEra::from(T::StakingRewardsPastErasMax::get());
```

Comment:
These range changes are needed but whatever else we use to calculate staking rewards is still TBD